python/GafferUI/ColorChooser.py

```diff
@@ -454,6 +451,121 @@ def __paintEvent( self, event ) :
 
 		self.__drawValue( painter )
 
+class _ColorFieldRowLayout( QtWidgets.QLayout ) :
+
+	def __init__( self ) :
+
+		QtWidgets.QLayout.__init__( self, None )
+
+		# No spacing between color field and sliders so the active component icons
+		# line up against the color field. Do add a little space between the sliders
+		# and the options button.
+		self.setSpacing( 0 )
+		self.__buttonSpacing = 4
+
+		self.__items = []
+
+	def addItem( self, item ) :
+
+		self.__items.append( item )
+
+	def count( self ) :
+
+		return len( self.__items )
+
+	def expandingDirections( self ) :
+
+		return QtCore.Qt.Orientations( QtCore.Qt.Orientation( 0 ) )
+
+	def itemAt( self, index ) :
+
+		return self.__items[index] if index >= 0 and index < len( self.__items ) else None
+
+	def takeAt( self, index ) :
+
+		return self.__items.pop( index ) if index >= 0 and index < len( self.__items ) else None
+
+	def minimumSize( self ) :
+
+		assert( len( self.__items ) == 3 )
+
+		return self.__size( self.__items[1].minimumSize(), self.__items[2].minimumSize() )
+
+	def maximumSize( self ) :
+
+		return QtCore.QSize( QtWidgets.QLayout.maximumSize( self ).width(), self.sizeHint().height() )
+
+	def sizeHint( self ) :
+
+		assert( len( self.__items ) == 3 )
+
+		return self.__size( self.__items[1].sizeHint(), self.__items[2].sizeHint() )
+
+	def setGeometry( self, rect ) :
+
+		assert( len( self.__items ) == 3 )
+
+		size1 = self.__items[1].sizeHint()
+		size2 = self.__items[2].sizeHint()
+
+		leftSize = 0
+		if not self.__items[0].isEmpty() :
+			self.__items[0].setGeometry( QtCore.QRect( rect.left(), rect.top(), size1.height(), size1.height() ) )
```

Comment:
Also happy to leave this to a follow-up PR after we get this and #5999 merged.

Reply:
I was a little shy of doing this at first because as you mentioned it somewhat opens the door to more configurability, but the heuristic works pretty well and is nice and simple so it seems worth doing. Done in https://github.com/GafferHQ/gaffer/pull/6003/commits/f448a562c750d9b3506ff563f046630129763832